### PR TITLE
feat: add optional line numbering to read_text_file tool

### PR DESCRIPTION
**Pull request by 0xErgod**
## Summary

Implements #60 - Adds an optional `with_line_numbers` parameter to the `read_text_file` tool to support line-numbered output for AI agents.

## Changes

- Added `with_line_numbers` optional boolean parameter to `ReadTextFile` struct
- Updated `read_text_file` service method to format output with line numbers when enabled
- Line numbers are right-aligned (6 digits) with pipe separator format: `     1 | content`
- Uses 1-based indexing for line numbers (matching standard text editor conventions)
- Maintains full backward compatibility (defaults to `false`)
- Added comprehensive unit tests covering various edge cases
- Updated CHANGELOG.md with feature description
- Updated capabilities.md documentation using mcp-discovery

## Motivation

Modern AI agents (like Codex and others) rely on line numbers to inject targeted patches to code. This feature allows agents to obtain all necessary context in a single tool invocation rather than making multiple calls, significantly improving efficiency for code modification tasks.

## Example Output

**Without line numbers** (default):
```
fn main() {
    println!("Hello");
}
```

**With line numbers** (`with_line_numbers: true`):
```
     1 | fn main() {
     2 |     println!("Hello");
     3 | }
```

## Test Plan

- All existing tests pass
- New tests added for line numbering feature:
  - Line numbers enabled
  - Line numbers disabled (backward compatibility)
  - Empty files
  - Single line files
  - Files without trailing newlines
  - Large files (1000+ lines)

## Testing

Run tests with:
```bash
cargo test --test test_fs_service test_read_text_file
```

All 6 new tests pass successfully.

## Validation

All project checks pass:
```bash
cargo make check
```

- Formatting check: Passed
- Clippy linting: Passed  
- All 123 tests: Passed
- Cargo check: Passed

[agent pull request]

### CODE REVIEW

File: src/tools/read_text_file.rs [hunk 1]
```diff
@@ -22,6 +23,11 @@ use crate::fs_service::FileSystemService;
 pub struct ReadTextFile {
     /// The path of the file to read.
     pub path: String,
+    /// Optional: Include line numbers in output (default: false).
+    /// When enabled, each line is prefixed with its line number (1-based).
+    /// Useful for AI agents that need to target specific lines for code patches.
```
Comment (hashemix) [inline src/tools/read_text_file.rs]: This makes it easier for AI systems to understand the format , more concise, precise, and clear.

```suggestion    
    /// When enabled, each line is prefixed with a right-aligned, 1-based line number
    /// Followed by a space, a vertical bar (`|`), and another space in the format: `   123 | <original line content>`
```

File: tests/test_fs_service.rs [hunk 2]
```diff
@@ -230,10 +230,82 @@ async fn test_unzip_file_non_existent() {
...
+        "line1\nline2\nline3",
+    );
+    let content = service.read_text_file(&file_path, false).await.unwrap();
+    assert_eq!(content, "line1\nline2\nline3");
+}
+
+#[tokio::test]
+async fn test_read_text_file_with_line_numbers_empty_file() {
+    let (temp_dir, service, _allowed_dirs) = setup_service(vec!["dir1".to_string()]);
+    let file_path = create_temp_file(temp_dir.join("dir1").as_path(), "empty.txt", "");
```
[the remark this answers is not in the document]
Comment (hashemix) [reply tests/test_fs_service.rs]: haha, that is ok, thanks for trying 👍

**Review by hashemix — CHANGES_REQUESTED** [review]
Hey @abderraouf-belalia  thanks for the updates, looks very good 🥇 

One last thing before I merge, ould you revert the changes to CHANGELOG.md? That file shouldn’t be modified in PRs.
Thanks

**Review by hashemix — APPROVED** [review]
